server/game/gameSystems/DiscardCardsFromHandSystem.ts

```diff
@@ -35,14 +36,25 @@ export class DiscardCardsFromHandSystem<TContext extends AbilityContext = Abilit
 
     public override getEffectMessage(context: TContext): [string, any[]] {
         const properties = this.generatePropertiesFromContext(context);
-        return ['make {0} {1}discard {2} cards', [properties.target, properties.random ? 'randomly ' : '', properties.amount]];
+        if (properties.discardingPlayerType === RelativePlayer.Self) {
+            return ['make {0} {1}discard {2} cards', [properties.target, properties.random ? 'randomly ' : '', properties.amount]];
+        }
+        return ['make {0} {1}discard {2} cards from {3}', [context.player, properties.random ? 'randomly ' : '', properties.amount, properties.target]];
     }
 
     public override canAffect(playerOrPlayers: Player | Player[], context: TContext, additionalProperties = {}, mustChangeGameState = GameStateChangeRequired.None): boolean {
         for (const player of Helpers.asArray(playerOrPlayers)) {
             const properties = this.generatePropertiesFromContext(context, additionalProperties);
             const availableHand = player.hand.filter((card) => properties.cardCondition(card, context) && EnumHelpers.cardTypeMatches(card.type, properties.cardTypeFilter));
 
+            // Select this player if its their own hand, or the active player from the context if its the 'opponent' choosing
+            const discardingPlayer = properties.discardingPlayerType === RelativePlayer.Self ? player : context.player;
+
+            // Since autoTarget is off any possible card in hand makes the hand 'canAffect' true
+            if (!discardingPlayer.autoSingleTarget && player.hand.length > 0) {
```

Comment:
This is not quite right - since the swu rules make it legal to target something for an effect even if there will be no game state change, we need to check the value of `mustChangeGameState` to determine whether the hand length is relevant.

I.e., if `mustChangeGameState === GameStateChangeRequired.None`, we should always return true here because it doesn't matter if any cards are being discarded. If it's `MustFullyOrPartiallyResolve` or `MustFullyResolve`, then we return true if there are any cards to discard (even if there are too few).

That isn't technically correct based on what those values mean but the "partial resolution" concept was only partly implemented, so for now we should match the below code which only checks `ustChangeGameState !== GameStateChangeRequired.None`. We can fix it all in one pass later when we finish implementing support for `MustFullyOrPartiallyResolve`.

If that doesn't make sense feel free to ping me in chat, sorry it's a bit of a mess atm

Reply:
Removed my change to this section, it was no longer needed (I believe?)